relay-kafka/src/producer/mod.rs

```diff
@@ -183,20 +187,36 @@ impl KafkaClient {
         variant: &str,
         payload: &[u8],
     ) -> Result<&str, ClientError> {
+        let now = Instant::now();
+
         let producer = self.producers.get(&topic).ok_or_else(|| {
             relay_log::error!(
                 "attempted to send message to {topic:?} using an unconfigured kafka producer",
             );
             ClientError::InvalidTopicName
         })?;
-        producer.send(key, headers, variant, payload)
+
+        if let Some(limiter) = self.rate_limiters.get(&topic) {
+            if limiter.try_increment(now, key, 1) < 1 {
+                metric!(
+                    counter(KafkaCounters::ProducerPartitionKeyRateLimit) += 1,
+                    variant = variant,
+                    topic = &producer.topic_name
+                );
+                return Ok(&producer.topic_name);
+            }
+        }
```

Comment:
This now drops data, instead of producing with a different key, I assume that's just temporary?

Reply:
yeah that was an oversight, also had to clean up some old code still